lib/active_admin/resource_controller/decorators.rb

```diff
@@ -48,18 +48,45 @@ def collection_decorator
         end
       end
 
+      class CollectionDecorator
```

Comment:
This class is essentially copied from Draper, so I would add a commennt on top of it about what it does, and include a permalink to the current version of this class inside Draper for proper attribution.